WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -58,4 +113,62 @@ open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
     open func queriesChanged() {
         // Subclasses should implement this
     }
+
+}
+
+// In ideal world, those wouldn't be extensions on `Encodable`/`Codable`, but rather refinements of
+// `QueryStore` in which `Store` conforms to `Codable` — then we would just change behavior _inside_ the `QueryStore`,
+// based on whether the type conforms to `Codable` or not.
+// Regretfully, Swift type system (or my Swift skills) does't allow to express that —
+// you can't just do `let codableSelf = self as? QueryStore<Codable, Query>` because
+// _Codable itself doesn't conform to Codable_.
```

Comment:
I think what you struggled with might come from these two methods not being really related to a store (and partly to `Codable` types). They're more like helpers around `JSONEncoder` to turn encoding/decoding into a one-liner.

How do you feel about making these two generic and don't mention "cached state" at all? I'd make them public and rename them to `saveJSON(at:)` and `loadJSON(from:)`